yeat/tests/test_cli.py

```diff
@@ -120,3 +121,56 @@ def test_custom_downsample_input(
     assert df.iloc[12]["sample_contigs"] == num_contigs
     assert df.iloc[13]["sample_contigs"] == largest_contig
     assert df.iloc[14]["sample_contigs"] == total_len
+
+
+@pytest.mark.parametrize("coverage", [("-1"), ("0")])
+def test_invalid_custom_coverage_1(coverage):
+    arglist = [
+        data_file("megahit.cfg"),
+        data_file("short_reads_1.fastq.gz"),
+        data_file("short_reads_2.fastq.gz"),
+        "--coverage",
+        coverage,
+    ]
+    with pytest.raises(SystemExit) as e:
+        args = cli.get_parser().parse_args(arglist)
+    assert isinstance(e.value.__context__, argparse.ArgumentError)
+    assert f"{coverage} is not a positive integer" in e.value.__context__.message
+
+
+@pytest.mark.parametrize("coverage", [("string"), ("3.14")])
+def test_invalid_custom_coverage_2(coverage):
+    arglist = [
+        data_file("megahit.cfg"),
+        data_file("short_reads_1.fastq.gz"),
+        data_file("short_reads_2.fastq.gz"),
+        "--coverage",
+        coverage,
+    ]
+    with pytest.raises(Exception, match=r"is not an integer"):
+        args = cli.get_parser().parse_args(arglist)
+
+
+@pytest.mark.long
+@pytest.mark.parametrize(
+    "coverage,num_contigs,largest_contig,total_len",
+    [("125", 56, 35168, 199940), ("150", 56, 35168, 199940)],
```

Comment:
You define 4 parameters, but 3 are invariant across all tests. It's ok to hard-code parameters that are the same for each test instance. Although, it would be interesting to include a test that resulted in a different number of contigs, total length, etc.

Reply:
> Although, it would be interesting to include a test that resulted in a different number of contigs, total length, etc.

Any suggestions on to test this? I completely agree and that is what I was hoping to do in this test; however, every time I tried to change the coverage, I was getting the same values of `num_contigs`, `largest_contig`, and `total_len`. The only time I was able to get something different was when I had set the coverage down to `1` and the number of contigs went down to 56 -> 31. The largest config stayed the same and the total length slightly decreased by a 100 bases.

I ultimately didn't go with 1x coverage because of its unrealistic input value (I could be wrong here).

Any suggestions would be welcomed. As of right now, I just have the different coverages and making sure that the values didn't change.